src/components/TaskItem/index.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from "react";
+
+import "./styles.scss";
+
+export class TaskItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      done: "notdone",
+      click: 0,
+    };
+  }
+
+  onTaskClickHandler = (e) => {
+    if (this.state.click % 2) {
+      this.setState((prev) => ({
+        ...prev,
+        done: "notdone",
+        click: ++prev.click,
+      }));
+    } else {
+      this.setState((prev) => ({
+        ...prev,
+        done: "done",
+        click: ++prev.click,
+      }));
```

Comment:
И когда done будет boolean не надо будет тут заниматься этими извращениями. 

```suggestion
      this.setState(({ done }) => ({
        done: !done,
      }));
```